fishtank.py

```diff
@@ -109,6 +116,33 @@ def move(self, walls, goblins):
             elif self.change_y < 0 and self.rect.top != goblin.rect.top:
                 self.rect.top = goblin.rect.bottom
 
+    def pick_target(self, possible_targets):
+        target_object = None
+
+        def look_within_cutoff(cutoff):
+            for target in possible_targets:
+                if abs(target.rect.x - self.rect.x) < cutoff and \
+                   abs(target.rect.y - self.rect.y) < cutoff:
+                    dist = distance(target.rect.x, target.rect.y, self.rect.x, self.rect.y)
+                    yield (dist, target)
+
+        for cutoff in (8, 128):
+            distances = look_within_cutoff(cutoff)
+            if distances:  # not empty
+                distances = sorted(distances)
+                try:
+                    target_object = distances[0][1]  # 0th (shortest dist), then the 1th element (object itself)
+                    break
+                except IndexError:
+                    continue
```

Comment:
This helped speed up choosing a target -- limiting one's search within a box.

`look_within_cutoff` filters out Coins that aren't `cutoff` or closer.

I do two searches with cutoff 8 and 128 before giving up.